torch/_dynamo/variables/functions.py

```diff
@@ -642,9 +642,24 @@ def as_python_constant(self):
 
 
 class TritonKernelVariable(VariableTracker):
-    def __init__(self, kernel, grid, **kwargs):
+    def __init__(self, kernel, kernel_idx, grid, **kwargs):
         super().__init__(**kwargs)
-        self.kernel = kernel
+
+        from torch._higher_order_ops.triton_kernel_wrap import (
+            add_kernel_to_table,
+            get_kernel_from_table,
+        )
+
+        if kernel is None:
+            assert kernel_idx is not None
+            self.kernel = get_kernel_from_table(kernel_idx)
+            self.kernel_idx = kernel_idx
+        else:
+            # TODO(oulgen): If kernel idx is not None and we had dedup,
+            # we should assert that kernel and idx matches
+            self.kernel = kernel
+            self.kernel_idx = add_kernel_to_table(kernel)
```

Comment:
How do you end up constructing a TritonKernelVariable where `kernel is None`?

Reply:
Uh this is crux from earlier versions of my testing, good catch, I will remove it